src/Adapter/Customer/Group/Validate/CustomerGroupValidator.php

```diff
@@ -0,0 +1,139 @@
+    {
+        $this->shopRepository = $shopRepository;
+    }
+
+    /**
+     * @param CustomerGroup $customerGroup
+     *
+     * @throws GroupConstraintException
+     *
+     * @return void
+     */
+    public function validate(CustomerGroup $customerGroup): void
+    {
+        $this->validateThereIsAtLeastOneShop($customerGroup->id_shop_list);
+        $this->validateShopsExists($customerGroup->id_shop_list);
+        $this->validateGroupNames($customerGroup->name);
+        $this->validatePriceDisplayMethod($customerGroup->price_display_method);
+    }
+
+    /**
+     * @param array $shopIds
+     *
+     * @throws GroupConstraintException
+     *
+     * @return void
+     */
+    private function validateThereIsAtLeastOneShop(array $shopIds): void
+    {
+        if (empty($shopIds)) {
+            throw new GroupConstraintException(
+                'Customer group must be associated with at least one shop',
+                GroupConstraintException::EMPTY_SHOP_LIST
+            );
+        }
+    }
+
+    /**
+     * @param array $shopIds
+     *
+     * @return void
+     */
+    private function validateShopsExists(array $shopIds): void
+    {
+        foreach ($shopIds as $shopId) {
+            $this->shopRepository->assertShopExists($shopId);
+        }
+    }
+
+    /**
+     * @param int $priceDisplayMethod
+     *
+     * @throws GroupConstraintException
+     *
+     * @return void
+     */
+    private function validatePriceDisplayMethod(int $priceDisplayMethod): void
+    {
+        switch ($priceDisplayMethod) {
+            case CustomerGroup::PRICE_DISPLAY_METHOD_TAX_INCL:
+            case CustomerGroup::PRICE_DISPLAY_METHOD_TAX_EXCL:
+                return;
+            default:
+                throw new GroupConstraintException(
+                    sprintf('Invalid price display method "%s"', $priceDisplayMethod),
+                    GroupConstraintException::INVALID_PRICE_DISPLAY_METHOD
+                );
+        }
+    }
+
+    /**
+     * @param string|string[] $names
+     *
+     * @throws GroupConstraintException
+     *
+     * @return void
+     */
+    private function validateGroupNames($names): void
+    {
+        if (is_array($names)) {
+            if (empty($names)) {
+                throw new GroupConstraintException(
+                    'Customer group name cannot be empty',
+                    GroupConstraintException::EMPTY_NAME
+                );
+            }
+            foreach ($names as $name) {
+                $this->validateGroupNames($name);
+            }
+        } else {
+            if (strlen($names) > 32 || false === preg_match('/^[^<>={}]*$/u', $names)) {
+                throw new GroupConstraintException(
+                    'Customer group name cannot contain the following characters: <>={}',
+                    GroupConstraintException::INVALID_NAME
+                );
+            }
+        }
+    }
+}
```

Comment:
I think you should split this function into two to avoid multiple inputs:
`private function validateGroupNames(array $names): void`
`private function validateGroupName(string $name): void`

In the ObjectModel this field can indeed have to types, but in the context of edition/creation the object should always be fetched without a specified language ID, meaning the localized fields are all in their array form

Reply:
Done